src/common/collections.rs

```diff
@@ -63,21 +63,49 @@ pub async fn do_list_aliases(
 }
 
 pub async fn do_list_snapshots(
-    toc: &TableOfContent,
+    dispatcher: &Dispatcher,
```

Comment:
Listing snapshot cannot be performed in an async fashion.

The caller has no mean to get the collection afterwards.